storage/chunker_test.go

```diff
@@ -151,7 +151,8 @@ func TestSha3ForCorrectness(t *testing.T) {
 	rawSha3Output := rawSha3.Sum(nil)
 
 	sha3FromMakeFunc := MakeHashFunc(SHA3Hash)()
```

Comment:
🌷 I know it's not part of this PR, but why not use constructor for Hasher instead of a func? If the func is needed maybe the builder can be extracted instead of this?

Reply:
The `SwarmHash` needs the length of the data it represents prepended as an 64-bit value. The `BMT` hash has this builtin, but the other types must be extended by `HashWithLength` to allow setting the length (`SetSpanBytes`, see `storage/swarmhasher.go`)